core/spec/models/spree/promotion/actions/create_line_items_spec.rb

```diff
@@ -48,6 +53,28 @@
         expect(line_item).not_to be_nil
         expect(line_item.quantity).to eq(3)
       end
+
+      it "doesn't try to add an item if it's out of stock" do
+        empty_stock(mug)
+        empty_stock(shirt)
+
+        expect(order.contents).to_not receive(:add)
+        action.perform(:order => order)
+      end
+    end
+  end
+
+  describe '#item_available?' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.